### PR TITLE
Add setEntityId() to entityForm

### PR DESCRIPTION
**Pull request by mattwire**
Overview
----------------------------------------
Add a function to set the entity ID - we can already get it via `getEntityID()` but not set.

Before
----------------------------------------
No function to set entity ID on the form.

After
----------------------------------------
Function to set the entity ID on the form.

Technical Details
----------------------------------------

Comments
----------------------------------------
I'm looking at making some improvements / conversion to contributionview form and this is a pre-requisite to doing it properly.

### CODE REVIEW

**Comment by seamuslee001** [comment]
@mattwire test failure

```
ok 867 - CRM_Core_Page_AJAXTest::testCheckAuthz
PHP Fatal error:  CRM_Member_Form_MembershipConfig and CRM_Core_Form_EntityFormTrait define the same property ($_id) in the composition of CRM_Member_Form_MembershipType. However, the definition differs and is considered incompatible. Class was composed in /home/jenkins/bknix-dfl/build/core-16020-2rekg/web/sites/all/modules/civicrm/CRM/Member/Form/MembershipType.php on line 23
PHP Stack trace:
PHP   1. {main}() /home/jenkins/bknix-dfl/civicrm-buildkit/extern/phpunit6/phpunit6.phar:0
PHP   2. PHPUnit\TextUI\Command::main() /home/jenkins/bknix-dfl/civicrm-buildkit/extern/phpunit6/phpunit6.phar:570
PHP   3. PHPUnit\TextUI\Command->run() phar:///home/jenkins/bknix-dfl/civicrm-buildkit/extern/phpunit6/phpunit6.phar/phpunit/TextUI/Command.php:148
PHP   4. PHPUnit\TextUI\TestRunner->doRun() phar:///home/jenkins/bknix-dfl/civicrm-buildkit/extern/phpunit6/phpunit6.phar/phpunit/TextUI/Command.php:195
PHP   5. PHPUnit\Framework\TestSuite->run() phar:///home/jenkins/bknix-dfl/civicrm-buildkit/extern/phpunit6/phpunit6.phar/phpunit/TextUI/TestRunner.php:545
PHP   6. PHPUnit\Framework\TestSuite->run() phar:///home/jenkins/bknix-dfl/civicrm-buildkit/extern/phpunit6/phpunit6.phar/phpunit/Framework/TestSuite.php:755
PHP   7. PHPUnit\Framework\TestSuite->run() phar:///home/jenkins/bknix-dfl/civicrm-buildkit/extern/phpunit6/phpunit6.phar/phpunit/Framework/TestSuite.php:755
PHP   8. CRM_Core_Page_HookTest->run() phar:///home/jenkins/bknix-dfl/civicrm-buildkit/extern/phpunit6/phpunit6.phar/phpunit/Framework/TestSuite.php:755
PHP   9. PHPUnit\Framework\TestResult->run() phar:///home/jenkins/bknix-dfl/civicrm-buildkit/extern/phpunit6/phpunit6.phar/phpunit/Framework/TestCase.php:894
PHP  10. CRM_Core_Page_HookTest->runBare() phar:///home/jenkins/bknix-dfl/civicrm-buildkit/extern/phpunit6/phpunit6.phar/phpunit/Framework/TestResult.php:698
PHP  11. CRM_Core_Page_HookTest->setUp() phar:///home/jenkins/bknix-dfl/civicrm-buildkit/extern/phpunit6/phpunit6.phar/phpunit/Framework/TestCase.php:935
PHP  12. is_subclass_of() /home/jenkins/bknix-dfl/build/core-16020-2rekg/web/sites/all/modules/civicrm/tests/phpunit/CRM/Core/Page/HookTest.php:59
PHP  13. spl_autoload_call() /home/jenkins/bknix-dfl/build/core-16020-2rekg/web/sites/all/modules/civicrm/tests/phpunit/CRM/Core/Page/HookTest.php:59
PHP  14. CRM_Core_ClassLoader->loadClass() /home/jenkins/bknix-dfl/build/core-16020-2rekg/web/sites/all/modules/civicrm/tests/phpunit/CRM/Core/Page/HookTest.php:59
PHP  15. require_once() /home/jenkins/bknix-dfl/build/core-16020-2rekg/web/sites/all/modules/civicrm/CRM/Core/ClassLoader.php:211

Fatal error: CRM_Member_Form_MembershipConfig and CRM_Core_Form_EntityFormTrait define the same property ($_id) in the composition of CRM_Member_Form_MembershipType. However, the definition differs and is considered incompatible. Class was composed in /home/jenkins/bknix-dfl/build/core-16020-2rekg/web/sites/all/modules/civicrm/CRM/Member/Form/MembershipType.php on line 23

Call Stack:
    0.0008     520528   1. {main}() /home/jenkins/bknix-dfl/civicrm-buildkit/extern/phpunit6/phpunit6.phar:0
    0.0371    9395176   2. PHPUnit\TextUI\Command::main() /home/jenkins/bknix-dfl/civicrm-buildkit/extern/phpunit6/phpunit6.phar:570
    0.0371    9403968   3. PHPUnit\TextUI\Command->run() phar:///home/jenkins/bknix-dfl/civicrm-buildkit/extern/phpunit6/phpunit6.phar/phpunit/TextUI/Command.php:148
    3.0942   93637184   4. PHPUnit\TextUI\TestRunner->doRun() phar:///home/jenkins/bknix-dfl/civicrm-buildkit/extern/phpunit6/phpunit6.phar/phpunit/TextUI/Command.php:195
    3.1215   93707464   5. PHPUnit\Framework\TestSuite->run() phar:///home/jenkins/bknix-dfl/civicrm-buildkit/extern/phpunit6/phpunit6.phar/phpunit/TextUI/TestRunner.php:545
  256.8721  162766984   6. PHPUnit\Framework\TestSuite->run() phar:///home/jenkins/bknix-dfl/civicrm-buildkit/extern/phpunit6/phpunit6.phar/phpunit/Framework/TestSuite.php:755
  359.3984  171715080   7. PHPUnit\Framework\TestSuite->run() phar:///home/jenkins/bknix-dfl/civicrm-buildkit/extern/phpunit6/phpunit6.phar/phpunit/Framework/TestSuite.php:755
  360.2296  167530888   8. CRM_Core_Page_HookTest->run() phar:///home/jenkins/bknix-dfl/civicrm-buildkit/extern/phpunit6/phpunit6.phar/phpunit/Framework/TestSuite.php:755
  360.2296  167530888   9. PHPUnit\Framework\TestResult->run() phar:///home/jenkins/bknix-dfl/civicrm-buildkit/extern/phpunit6/phpunit6.phar/phpunit/Framework/TestCase.php:894
  360.2297  167531728  10. CRM_Core_Page_HookTest->runBare() phar:///home/jenkins/bknix-dfl/civicrm-buildkit/extern/phpunit6/phpunit6.phar/phpunit/Framework/TestResult.php:698
  360.2297  167548192  11. CRM_Core_Page_HookTest->setUp() phar:///home/jenkins/bknix-dfl/civicrm-buildkit/extern/phpunit6/phpunit6.phar/phpunit/Framework/TestCase.php:935
  360.2964  178337880  12. is_subclass_of() /home/jenkins/bknix-dfl/build/core-16020-2rekg/web/sites/all/modules/civicrm/tests/phpunit/CRM/Core/Page/HookTest.php:59
  360.2964  178337936  13. spl_autoload_call() /home/jenkins/bknix-dfl/build/core-16020-2rekg/web/sites/all/modules/civicrm/tests/phpunit/CRM/Core/Page/HookTest.php:59
  360.2964  178338056  14. CRM_Core_ClassLoader->loadClass() /home/jenkins/bknix-dfl/build/core-16020-2rekg/web/sites/all/modules/civicrm/tests/phpunit/CRM/Core/Page/HookTest.php:59
  360.2967  178441360  15. require_once('/home/jenkins/bknix-dfl/build/core-16020-2rekg/web/sites/all/modules/civicrm/CRM/Member/Form/MembershipType.php') /home/jenkins/bknix-dfl/build/core-16020-2rekg/web/sites/all/modules/civicrm/CRM/Core/ClassLoader.php:211
```

**Comment by mattwire** [comment]
test this please

**Comment by mattwire** [comment]
test this please

**Comment by mattwire** [comment]
test this please

**Comment by mattwire** [comment]
test this please

**Comment by mattwire** [comment]
test this please

**Comment by mattwire** [comment]
Hey @seamuslee001 is this failing because of the PropertyBag deprecated warnings?

**Comment by seamuslee001** [comment]
@mattwire no

```
Fatal error: CRM_Member_Form and CRM_Core_Form_EntityFormTrait define the same property ($_id) in the composition of CRM_Member_Form. However, the definition differs and is considered incompatible. Class was composed in /home/jenkins/bknix-dfl/build/core-16020-5whfr/web/sites/all/modules/civicrm/CRM/Member/Form.php on line 22
```

File: CRM/Core/Form/EntityFormTrait.php [hunk 1]
```diff
@@ -16,6 +16,13 @@
  */
 trait CRM_Core_Form_EntityFormTrait {
 
+  /**
+   * The id of the object being edited / created.
+   *
+   * @var int
+   */
+  protected $_id;
```
Comment (seamuslee001) [inline CRM/Core/Form/EntityFormTrait.php]: @mattwire is there a reason why your making this protected rather than public?

File: CRM/Member/Form/MembershipConfig.php [hunk 2]
```diff
@@ -28,7 +28,7 @@ class CRM_Member_Form_MembershipConfig extends CRM_Core_Form {
    *
    * @var int
    */
-  public $_id;
+  protected $_id;
```
Comment (seamuslee001) [inline CRM/Member/Form/MembershipConfig.php]: @mattwire can you revert this change?

**Comment by seamuslee001** [comment]
@mattwire as your changing the visibility of $_id in CRM_Admin_Form then you need to make these changes as well

```
diff --git a/CRM/Admin/Form/Job.php b/CRM/Admin/Form/Job.php
index 8a8679b5b4..3009a5b730 100644
--- a/CRM/Admin/Form/Job.php
+++ b/CRM/Admin/Form/Job.php
@@ -19,7 +19,7 @@
  * Class for configuring jobs.
  */
 class CRM_Admin_Form_Job extends CRM_Admin_Form {
-  protected $_id = NULL;
+  public $_id = NULL;

   public function preProcess() {

diff --git a/CRM/Admin/Form/LabelFormats.php b/CRM/Admin/Form/LabelFormats.php
index 1713fd910b..9c60fe8de7 100644
--- a/CRM/Admin/Form/LabelFormats.php
+++ b/CRM/Admin/Form/LabelFormats.php
@@ -41,7 +41,7 @@ class CRM_Admin_Form_LabelFormats extends CRM_Admin_Form {
    * Label Format ID.
    * @var int
    */
-  protected $_id = NULL;
+  public $_id = NULL;

   /**
    * Group name, label format or name badge
diff --git a/CRM/Admin/Form/PaymentProcessorType.php b/CRM/Admin/Form/PaymentProcessorType.php
index 571aa957ac..89adf8ac7b 100644
--- a/CRM/Admin/Form/PaymentProcessorType.php
+++ b/CRM/Admin/Form/PaymentProcessorType.php
@@ -19,7 +19,7 @@
  * This class generates form components for Location Type.
  */
 class CRM_Admin_Form_PaymentProcessorType extends CRM_Admin_Form {
-  protected $_id = NULL;
+  public $_id = NULL;

   protected $_fields = NULL;

diff --git a/CRM/Admin/Form/PdfFormats.php b/CRM/Admin/Form/PdfFormats.php
index 8588c55516..0ec9042d58 100644
--- a/CRM/Admin/Form/PdfFormats.php
+++ b/CRM/Admin/Form/PdfFormats.php
@@ -41,7 +41,7 @@ class CRM_Admin_Form_PdfFormats extends CRM_Admin_Form {
    * PDF Page Format ID.
    * @var int
    */
-  protected $_id = NULL;
+  public $_id = NULL;

   /**
    * Build the form object.
diff --git a/CRM/Admin/Form/ScheduleReminders.php b/CRM/Admin/Form/ScheduleReminders.php
index a845b5c644..e0fd77de3c 100644
--- a/CRM/Admin/Form/ScheduleReminders.php
+++ b/CRM/Admin/Form/ScheduleReminders.php
@@ -24,7 +24,7 @@ class CRM_Admin_Form_ScheduleReminders extends CRM_Admin_Form {
    * Scheduled Reminder ID.
    * @var int
    */
-  protected $_id = NULL;
+  public $_id = NULL;

   public $_freqUnits;

```

**Comment by seamuslee001** [comment]
@mattwire same test is still failing `CRM_Core_Page_AJAXTest::testFormsCallBuildFormOnce`

**Comment by mattwire** [comment]
@seamuslee001 Think we're finally there with this one!

**Comment by eileenmcnaughton** [comment]
I'm just removing merge ready because that doesn't mean the way you are using it - it you give it to another person's Pr when you are ready to merge it but just want to give it few days cool off inn  case someone else has a comment. Merge on  pass is when  you review another person's PR & it can be merged as soon as tests pass.

There are 2 things  I'd want to check before merging this 

1) We should avoid making $id public if we can - we should aim for protected in general so we can better check what interacts with our code. 
2) sign setEntityId vs setEntityID() - I'll have to check but I feel we have been doing things the latter way increasingly

**Comment by eileenmcnaughton** [comment]
Oh - the latter point -we have getEntityID & this adds setEntityId - I think it has to be caps then

**Comment by mattwire** [comment]
> We should avoid making $id public if we can - we should aim for protected in general so we can better check what interacts with our code.

Believe me I tried! Unfortunately the membership classes access the entity ID which (currently) require it to be public as they access the id property directly.

> sign setEntityId vs setEntityID() - I'll have to check but I feel we have been doing things the latter way increasingly
> Oh - the latter point -we have getEntityID & this adds setEntityId - I think it has to be caps then

I'm confused. We have `getEntityId` not `getEntityID` which is why I put it as `setEntityId`. I really don't mind which way though so please confirm which you'd prefer @eileenmcnaughton

**Comment by eileenmcnaughton** [comment]
Ah - I based that on reading the PR not the code - PR comment is wrong it's entityId as you say.

OK - I'm happy to merge then

**Comment by mattwire** [comment]
Thanks @eileenmcnaughton if I get chance I'll try and follow up sometime with some work on the membership classes to use this function and then we could probably switch back to a protected property.

**Comment by eileenmcnaughton** [comment]
@mattwire yeah that would be good - they just need to call the sets & gets
